flake: post-kueue-push-images: failed to authorize: DeadlineExceeded: context deadline exceede

<!--
Please use this template for reporting flaky tests.
Links to specific failures in Prow are appreciated.
-->

**Which test is flaking?**:

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/post-kueue-push-images/2070349939271012352
**Link to failed CI job or steps to reproduce locally**:

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/post-kueue-push-images/2070349939271012352
**Failure message or logs**:
```
#32 [auth] k8s-staging-images/kueue/kueue-priority-booster:pull,push token for us-central1-docker.pkg.dev
#32 DONE 0.0s
#31 exporting to image
#31 exporting to image
#31 pushing layers 1.3s done
#31 pushing layers 1.4s done
#31 pushing manifest for us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue-priority-booster:v0.19.0-devel-237-g8ea8b61e6@sha256:1c4d3a3b563b5f4021d01c0e073ae68c4beba54c2e149f8ddd373457c4235412
#31 pushing manifest for us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue-populator:v0.19.0-devel-237-g8ea8b61e6@sha256:4f20e96db0705ebdf2ef2912721f85a0d106345b93ef8f280843a64889cc5387
#27 pushing layers 60.2s done
#27 ERROR: failed to push us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueueviz-backend:v0.19.0-devel-237-g8ea8b61e6: failed to authorize: DeadlineExceeded: context deadline exceeded
#31 pushing manifest for us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue-priority-booster:v0.19.0-devel-237-g8ea8b61e6@sha256:1c4d3a3b563b5f4021d01c0e073ae68c4beba54c2e149f8ddd373457c4235412 1.1s done
------
 > exporting to image:
------
ERROR: failed to build: failed to solve: DeadlineExceeded: context deadline exceeded
make[1]: *** [Makefile:442: kueueviz-image-build] Error 1
make[1]: *** Waiting for unfinished jobs....
#31 pushing manifest for us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue-populator:v0.19.0-devel-237-g8ea8b61e6@sha256:4f20e96db0705ebdf2ef2912721f85a0d106345b93ef8f280843a64889cc5387 1.5s done
#31 pushing layers 0.6s done
#31 pushing manifest for us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue-priority-booster:main@sha256:1c4d3a3b563b5f4021d01c0e073ae68c4beba54c2e149f8ddd373457c4235412
#31 pushing layers 0.2s done
#31 pushing manifest for us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue-populator:main@sha256:4f20e96db0705ebdf2ef2912721f85a0d106345b93ef8f280843a64889cc5387
#31 pushing manifest for us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue-populator:main@sha256:4f20e96db0705ebdf2ef2912721f85a0d106345b93ef8f280843a64889cc5387 0.3s done
#31 DONE 6.3s
#31 pushing manifest for us-central1-docker.pkg.dev/k8s-staging-images/kueue/kueue-priority-booster:main@sha256:1c4d3a3b563b5f4021d01c0e073ae68c4beba54c2e149f8ddd373457c4235412 0.7s done
#31 DONE 6.7s
BUILD FAILURE: Build step failure: build step 0 "gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20260127-c1affcc8de" failed: step exited with non-zero status: 2
ERROR: (gcloud.builds.submit) build 64ab3553-243a-41d2-8c8c-9b65acc7248d completed with status "FAILURE"
make[2]: Leaving directory '/workspace/cmd/experimental/kueue-populator'
make[2]: Leaving directory '/workspace/cmd/experimental/kueue-priority-booster'
#23 ...
```

**Anything else we need to know?**:

--- a/Makefile
+++ b/Makefile
@@ -73,6 +73,8 @@ CGO_ENABLED ?= 0
 
 YAML_PROCESSOR_LOG_LEVEL ?= info
 
+IMAGE_PUSH_RETRY = $(PROJECT_DIR)/hack/testing/retry.sh --attempts 7 --delay 2 --exponential --stream --continue-if "grep -qiE 'context deadline exceeded' {output}" -- env
+
 # Setting SHELL to bash allows bash commands to be executed by recipes.
 # This is a requirement for 'setup-envtest.sh' in the test target.
 # Options are set to exit when a recipe line exits non-zero or a piped command fails.
@@ -226,6 +228,7 @@ image-local-build:
 
 # Build the multiplatform container image locally and push to repo.
 .PHONY: image-local-push
+image-local-push: IMAGE_BUILD_CMD := $(IMAGE_PUSH_RETRY) $(IMAGE_BUILD_CMD)
 image-local-push: PUSH=--push
 image-local-push: image-local-build
 
@@ -253,6 +256,7 @@ image-pushing-postsubmit:
 	$(MAKE) -j5 image-push helm-chart-push kueueviz-image-push kueue-populator-image-push kueue-priority-booster-image-push
 
 .PHONY: image-push
+image-push: IMAGE_BUILD_CMD := $(IMAGE_PUSH_RETRY) $(IMAGE_BUILD_CMD)
 image-push: PUSH=--push
 image-push: image-build
 
@@ -401,7 +405,7 @@ update-security-insights: yq
 # Developers don't need to build this image, as it will be available as us-central1-docker.pkg.dev/k8s-staging-images/kueue/debug
 .PHONY: debug-image-push
 debug-image-push: ## Build and push the debug image to the registry
-	$(IMAGE_BUILD_CMD) \
+	$(IMAGE_PUSH_RETRY) $(IMAGE_BUILD_CMD) \
 		-t $(IMAGE_REGISTRY)/debug:$(GIT_TAG) \
 		-t $(IMAGE_REGISTRY)/debug:$(RELEASE_BRANCH) \
 		--platform=$(PLATFORMS) \
@@ -426,6 +430,7 @@ importer-image-build:
 		-f ./cmd/importer/Dockerfile ./
 
 .PHONY: importer-image-push
+importer-image-push: IMAGE_BUILD_CMD := $(IMAGE_PUSH_RETRY) $(IMAGE_BUILD_CMD)
 importer-image-push: PUSH=--push
 importer-image-push: importer-image-build
 
@@ -458,6 +463,7 @@ kueueviz-image-build:
 		-f ./cmd/kueueviz/frontend/Dockerfile ./cmd/kueueviz/frontend
 
 .PHONY: kueueviz-image-push
+kueueviz-image-push: IMAGE_BUILD_CMD := $(IMAGE_PUSH_RETRY) $(IMAGE_BUILD_CMD)
 kueueviz-image-push: PUSH=--push
 kueueviz-image-push: kueueviz-image-build
 
@@ -481,6 +487,7 @@ kueue-populator-image-build:
 		IMAGE_BUILD_EXTRA_OPTS="$(IMAGE_BUILD_EXTRA_OPTS) -t $(IMAGE_REPO_KUEUE_POPULATOR):$(RELEASE_BRANCH)"
 
 .PHONY: kueue-populator-image-push
+kueue-populator-image-push: IMAGE_BUILD_CMD := $(IMAGE_PUSH_RETRY) $(IMAGE_BUILD_CMD)
 kueue-populator-image-push: PUSH=--push
 kueue-populator-image-push: kueue-populator-image-build
 
@@ -504,6 +511,7 @@ kueue-priority-booster-image-build:
 		IMAGE_BUILD_EXTRA_OPTS="$(IMAGE_BUILD_EXTRA_OPTS) -t $(IMAGE_REPO_KUEUE_PRIORITY_BOOSTER):$(RELEASE_BRANCH)"
 
 .PHONY: kueue-priority-booster-image-push
+kueue-priority-booster-image-push: IMAGE_BUILD_CMD := $(IMAGE_PUSH_RETRY) $(IMAGE_BUILD_CMD)
 kueue-priority-booster-image-push: PUSH=--push
 kueue-priority-booster-image-push: kueue-priority-booster-image-build
 
@@ -555,6 +563,7 @@ ray-project-mini-image-build:
 		-f ./hack/testing/ray-mini/Dockerfile ./
 
 .PHONY: ray-project-mini-image-build-push
+ray-project-mini-image-build-push: IMAGE_BUILD_CMD := $(IMAGE_PUSH_RETRY) $(IMAGE_BUILD_CMD)
 ray-project-mini-image-build-push: PUSH=--push
 ray-project-mini-image-build-push: ray-project-mini-image-build
 

